core/src/main/java/zipkin2/reporter/internal/AsyncReporter.java

```diff
@@ -216,18 +220,15 @@ void startFlusherThread() {
       flushThread.start();
     }
 
+    @SuppressWarnings("unchecked")
     @Override public void report(S next) {
       if (next == null) throw new NullPointerException("span == null");
       // Lazy start so that reporters never used don't spawn threads
       if (started.compareAndSet(false, true)) startFlusherThread();
       metrics.incrementSpans(1);
-      int nextSizeInBytes = encoder.sizeInBytes(next);
-      int messageSizeOfNextSpan = sender.messageSizeInBytes(nextSizeInBytes);
-      metrics.incrementSpanBytes(nextSizeInBytes);
-      if (closed.get() ||
-        // don't enqueue something larger than we can drain
-        messageSizeOfNextSpan > messageMaxBytes ||
-        !pending.offer(next, nextSizeInBytes)) {
+
+      // enqueue now and filter our when we drain
+      if (closed.get() || !pending.offer(next)) {
```

Comment:
fyi I think this should be fine, even if the logic was moved, because in armeria it is about "block calls inside" a method, and this is still inside the same method https://github.com/line/armeria/blob/333a16f3e617e7453b354f419c2b58893831bd71/brave/brave6/src/main/java/com/linecorp/armeria/common/brave/BraveBlockHoundIntegration.java#L35